pkg/apis/storageos/v1/storageoscluster_types.go

```diff
@@ -291,6 +296,17 @@ func (s StorageOSClusterSpec) GetCSILivenessProbeImage() string {
 	return CSIv1LivenessProbeContainerImage
 }
 
+// GetHyperkubeImage returns hyperkube container image for a given k8s version.
+// If an image is set explicitly in the cluster configuration, that image is
+// returned.
+func (s StorageOSClusterSpec) GetHyperkubeImage(k8sVersion string) string {
+	if s.Images.HyperkubeContainer != "" {
+		return s.Images.HyperkubeContainer
+	}
+	// Add version prefix "v" in the tag.
+	return fmt.Sprintf("%s:v%s", DefaultHyperkubeContainerRegistry, k8sVersion)
```

Comment:
I found that for many clusters the image tag :k8sVersion results in a pull failure as the image doesn't exist.

For instance the following cluster definition tries to pull `gcr.io/google_containers/hyperkube:v1.12.6-eks-d69f1b` as the `k8sVersion` is `1.12.6-eks-d69f1b`. Image that doesn't exist. 

```
serverVersion:
  buildDate: 2019-02-28T20:26:10Z
  compiler: gc
  gitCommit: d69f1bf3669bf00b7f4a758e978e0e7a1e3a68f7
  gitTreeState: clean
  gitVersion: v1.12.6-eks-d69f1b
  goVersion: go1.10.8
  major: "1"
  minor: 12+
  platform: linux/amd64
```

Reply:
ah! So those are commit hashes. We can trim the version and extract the first tri-dotted part of the version.
I'll add something to handle this with tests.
If you find more variants, please share about them.

Reply:
This issue can be mitigated with documentation by defining an explicit image in the CR for specific providers in case that you consider that the `k8sVersion` variable should follow the standard.

Reply:
I found that Rancher clusters deployed with a Driver also use the hash of the commit. Without using a driver, the k8sVersion is as expected
```
serverVersion:
  buildDate: 2019-06-19T20:50:07Z
  compiler: gc
  gitCommit: fcbc1d20b6bca1936c0317743055ac75aef608ce
  gitTreeState: clean
  gitVersion: v1.13.6-gke.13
  goVersion: go1.11.5b4
  major: "1"
  minor: 13+
  platform: linux/amd64
```

Reply:
I don't know if its feasible but the `major` + `minor` seems more stable for what we are trying to do.